src/main/java/com/streamr/client/StreamrRESTClient.java

```diff
@@ -68,16 +70,26 @@ public StreamrRESTClient(StreamrClientOptions options) {
 
     private <T> T get(HttpUrl url, JsonAdapter<T> adapter) throws IOException {
         Request.Builder builder = new Request.Builder().url(url);
-        Request request = addAuthenticationHeader(builder).build();
-        return execute(request, adapter);
+        Request request = addAuthenticationHeader(builder, false).build();
+        try {
```

Comment:
The retry logic could be wrapped into something like `executeWithRetry` to avoid repetition